### PR TITLE
Add analysis for concept exercise `marios-marvellous-lasagna`

### LINKED ISSUE #14: Write analysis for concept exercise `marios-marvellous-lasagna`
[From `design.md`](https://github.com/exercism/elm/blob/main/exercises/concept/marios-marvellous-lasagna/.meta/design.md):

This exercise could benefit from the following rules added to the the analyzer:
- make sure that `let` is used somewhere

### PR DESCRIPTION
**Pull request by jiegillet**
Closes #14.

[Sister PR here](https://github.com/exercism/website-copy/pull/2227).

I only added one rule for checking if a `let` has been used (I added the possibility in `Analyzer.functionCalls`).

Is it worth adding a rule checking that there are two expressions defined in the let block or what their names are? It wouldn't be very hard, but there isn't that much benefit, unless we want to make sure that they know that you can also define functions in there, not just values.

### CODE REVIEW

File: src/Analyzer.elm
Comment (ceddlyburge) [inline src/Analyzer.elm:148]: Can we use the documentation function comment style that elm-format likes here?

```{-| .... -}```

File: src/Analyzer.elm
Comment (ceddlyburge) [inline src/Analyzer.elm:177]: Can we use the documentation function comment style that elm-format likes here?

```{-| .... -}```

File: src/Exercise/MariosMarvellousLasagna.elm
Comment (ceddlyburge) [inline src/Exercise/MariosMarvellousLasagna.elm:19]: Can we use the documentation function comment style that elm-format likes here?

```{-| .... -}```

File: test_data/two-fer/imperfect_solution/expected_analysis.json
Comment (ceddlyburge) [inline test_data/two-fer/imperfect_solution/expected_analysis.json:1]: Is this due to the update in elm-review?

**Review by ceddlyburge — APPROVED** [review]

File: test_data/two-fer/imperfect_solution/expected_analysis.json
Comment (jiegillet) [reply test_data/two-fer/imperfect_solution/expected_analysis.json:1]: Yes! 
Some messages changed the wording a bit, and the order had changed for some reason (but I can't be mad, it's not like the order is guaranteed by the elm-review API) so I added a secondary order to our comments to help with that. 
I suppose approach would be to use `jq` to order the fields before comparing them but that might be a bit tricky too.

File: src/Analyzer.elm
Comment (jiegillet) [reply src/Analyzer.elm:148]: Ah, I never noticed it did that

**Comment by jiegillet** [comment]
All done!

@exercism/maintainers-admin  could we get an approval?

**Review by ErikSchierboom — APPROVED** [review]
LGTM

**Pull request merged**
